api/auth.go

```diff
@@ -61,16 +56,65 @@ func (a *API) withToken(ctx context.Context, w http.ResponseWriter, r *http.Requ
 		}
 	}
 
-	log = log.WithFields(logrus.Fields{
+	log.WithFields(logrus.Fields{
 		"claims_id":    claims.ID,
 		"claims_email": claims.Email,
 		"roles":        roles,
 		"is_admin":     isAdmin,
-	})
+	}).Info("successfully parsed claims")
 
-	log.Info("successfully parsed claims")
 	ctx = gcontext.WithAdminFlag(ctx, isAdmin)
-	ctx = gcontext.WithLogger(ctx, log)
+	ctx = gcontext.WithToken(ctx, token)
+	return ctx, nil
+}
 
-	return gcontext.WithToken(ctx, token)
+func authRequired(w http.ResponseWriter, r *http.Request) (context.Context, error) {
+	ctx := r.Context()
+	claims := gcontext.GetClaims(ctx)
+	if claims == nil {
+		return nil, unauthorizedError("No claims provided")
+	}
+
+	return ctx, nil
+}
+
+func adminRequired(w http.ResponseWriter, r *http.Request) (context.Context, error) {
+	ctx := r.Context()
+	claims := gcontext.GetClaims(ctx)
+	isAdmin := gcontext.IsAdmin(ctx)
+
+	if claims == nil || !isAdmin {
+		return nil, unauthorizedError("Admin permissions required")
+	}
+
+	logEntrySetField(r, "admin_id", claims.ID)
```

Comment:
doesn't this return a logger? Or does it mutate the logger that is being used for _this_ request?